extractor/src/main/scala/com/digitalasset/extractor/json/JsonConverters.scala

```diff
@@ -89,10 +96,22 @@ object JsonConverters {
     }
   }
 
-  implicit def taggedEncoder[A: Encoder, T]: Encoder[scalaz.@@[A, T]] =
-    Encoder[A].contramap(scalaz.Tag.unwrap)
-  implicit def taggedDecoder[A: Decoder, T]: Decoder[scalaz.@@[A, T]] =
-    Decoder[A].map(scalaz.Tag.apply)
+  implicit def taggedEncoder[A: Encoder, T]: Encoder[A @@ T] =
+    scalaz.Tag.subst(Encoder[A])
+  implicit def taggedDecoder[A: Decoder, T]: Decoder[A @@ T] =
+    scalaz.Tag.subst(Decoder[A])
+
+  implicit val nameEncoder: Encoder[Ref.Name] =
+    Encoder[String].contramap(identity)
+  implicit val partyEncoder: Encoder[Ref.Party] =
+    Encoder[String].contramap(identity)
+
+  // TODO SC this matches the prior behavior of JSON-ing the dates and timestamps
+  // as days-since and micros-since epoch, but maybe we'd like something else?
```

Comment:
cc @bitonic @remyhaemmerle-da @leo-da for JSON-pinions

Reply:
Yes, i think we should change this to use string representations, as we did with the old JSON-through-LfTypeEncoding. Feel free to change this in the next PR though.